x/src/test.rs

```diff
@@ -17,21 +21,71 @@ pub struct Args {
     #[structopt(long, short)]
     /// Only run unit tests
     unit: bool,
+
+    #[structopt(long)]
+    /// Do not fast fail the run if tests (or test executables) fail.
+    no_fail_fast: bool,
+    #[structopt(long)]
+    /// Do not run tests, only print their names.
+    no_run: bool,
+
+    #[structopt(long, parse(from_os_str))]
+    lcovdir: Option<OsString>,
+    #[structopt(long, parse(from_os_str))]
+    htmlcovdir: Option<OsString>,
     #[structopt(name = "TESTNAME", parse(from_os_str))]
     testname: Option<OsString>,
+
     #[structopt(name = "ARGS", parse(from_os_str), last = true)]
     args: Vec<OsString>,
 }
 
 pub fn run(mut args: Args, config: Config) -> Result<()> {
     args.args.extend(args.testname.clone());
 
-    let cmd = CargoCommand::Test(&args.args);
+    //if we will be be running coverage, set the environment arguments we'll force
+    let runcov = args.lcovdir.is_some() || args.htmlcovdir.is_some();
+    let coverageflags: &[(&str, &str)] = &[
+        //A way to use -Z (unstable) flags with the stable compiler. See below.
+        ("RUSTC_BOOTSTRAP", "1"),
+        //Recommend setting for grcov, avoids using the cargo cache.
+        ("CARGO_INCREMENTAL", "0"),
+        //recommend flags for use with grcov, with these flags removed: -Copt-level=0, -Clink-dead-code.
+        //for more info see:  https://github.com/mozilla/grcov#example-how-to-generate-gcda-fiels-for-a-rust-project
+        (
+            "RUSTFLAGS",
+            "-Zprofile -Ccodegen-units=1 -Coverflow-checks=off -Zno-landing-pads",
+        ),
+    ];
+
+    let env_vars = if runcov { coverageflags } else { &[] };
+
+    if runcov {
+        let mut clean_cmd = Command::new("cargo");
+        clean_cmd.arg("clean");
+        clean_cmd.output()?;
+        info!("Since we are running coverage, we must run cargo clean first");
+    }
```

Comment:
torn on doing this here, maybe add a flag?   maybe try something less damaging to remove just the grcov data files, bit of testing to figure that out.